taskvine/src/bindings/python3/taskvine.binding.py

```diff
@@ -1574,25 +1580,104 @@ def submit(self, task):
     #
     # @param self   Reference to the current manager object.
-    # @param name   Name of the duty to be removed.
-    def remove_duty(self, name):
-        del self._duty_table[name]
-        vine_manager_remove_duty(self._taskvine, "duty_coprocess:" + name)
+    # @param name   Name of the library to be removed.
+    def remove_library(self, name):
+        del self._library_table[name]
+        vine_manager_remove_library(self._taskvine, name)
+
+    # Turn a list of python functions into a Library
+    #
+    #
+    # @param self            Reference to the current manager object.
+    # @param name            Name of the Library to be created
+    # @param function_list   List of all functions to be included in the library
+    def create_library_from_functions(self, name, *function_list):
+        # ensure poncho python library is available
+        if poncho_available == False:
+            raise TypeError("The poncho module is not available. Cannot create Library.")
+        # positional arguments are the list of functions to include in the library
+        # create a unique hash of a combination of function names and bodies
+        functions_hash = package_serverize.generate_functions_hash(function_list)
+
+        # create path for caching library code and environment based on function hash
+        library_cache_path = f"./vine-library-cache/{functions_hash}"
+        library_code_path = f"{library_cache_path}/library_code.py"
+        library_env_path = f"{library_cache_path}/library_env.tar.gz"
+
+        # library cache folder doesn't exist, create it
+        if not os.path.exists("vine-library-cache"):
+            os.mkdir("vine-library-cache")
+        # if the library code and environment exist, move on to creating the Library Task
+        if os.path.isfile(library_code_path) and os.path.isfile(library_env_path):
+            pass
+        else:
+            # check if the cache entry exits, and create it if it doesn't exist 
+            if not os.path.exists(library_cache_path):
+                os.mkdir(library_cache_path)
+            print("No cached Library code and environment found, regenerating...")
+            # create library code and environment
+            package_serverize.serverize_library_from_code(library_cache_path, function_list, name)
+            # enable correct permissions for library code
+            os.chmod(library_code_path, 0o775)
+        
+        # create Task to execute the Library
+        t = LibraryTask("./poncho_package_run -e package python ./library_code.py", name)
+        # declare the environment
+        f = self.declare_poncho(self.declare_file(library_env_path))
+        t.add_input(f, "package")
+        # declare the library code as an input
+        f = self.declare_file(library_code_path)
+        t.add_input(f, "library_code.py")
+        # add poncho_package_run to be able to execute the Library
+        f = self.declare_file(shutil.which('poncho_package_run'))
```

Comment:
poncho_package_run does not need to be added anymore (as of yesterday), as poncho tarballs now have it included.